libsolutil/CommonIO.cpp

```diff
@@ -43,9 +43,17 @@ namespace
 template <typename T>
 inline T readFile(std::string const& _file)
 {
+	assertThrow(boost::filesystem::exists(_file), FileNotFound, _file);
+
+	// ifstream does not always fail when the path leads to a directory. Instead it might succeed
+	// with tellg() returning a nonsensical value so that std::length_error gets raised in resize().
+	assertThrow(boost::filesystem::is_regular_file(_file), NotAFile, _file);
```

Comment:
What if it's a symlink?

Reply:
Then it still works. `is_regular_file()` will follow the symlink.

I was also worried about this exact case which is why I included a test in this PR. Test only covers `readFileAsString()` but `CommandLineInterface` has the same checks (`exists()` + `is_regular_file()`) so it should go through them too. Just in case I tested this manually and `solc` does successfully compile a file specified via a symlink.